lib/model/model_test.go

```diff
@@ -2502,6 +2469,78 @@ func TestCustomMarkerName(t *testing.T) {
 	}
 }
 
+func TestIssue4475(t *testing.T) {
+	scanned := filepath.Join("baz", "scanned")
+	unscanned := filepath.Join("baz", "unscanned")
+	defer func() {
+		defaultFs.Remove(scanned)
+		defaultFs.Remove(unscanned)
+	}()
+
+	fd, err := defaultFs.Create(scanned)
+	if err != nil {
+		t.Fatal(err)
+		return
+	}
+	fd.Close()
+
+	dbi := db.OpenMemory()
+	m := NewModel(defaultConfig, protocol.LocalDeviceID, "syncthing", "dev", dbi, nil)
+	m.AddFolder(defaultFolderConfig)
+	m.StartFolder("default")
+	m.ServeBackground()
+	defer m.Stop()
+	m.ScanFolder("default")
+
+	// Scenario: Folder is deleted on the remote, while locally two files exist
+	// in it. One is already scanned, the other not.
+
+	fd, err = defaultFs.Create(unscanned)
+	if err != nil {
+		t.Fatal(err)
+		return
+	}
+	fd.Close()
+
+	dir, ok := m.CurrentFolderFile("default", "baz")
+	if !ok {
+		t.Fatalf("Can't get dir \"baz\" after initial scan")
+	}
+
+	dir.Deleted = true
+	version := dir.Version.Update(device1.Short()).Update(device1.Short())
```

Comment:
Why twice?

Reply:
because the original version does not have device1 in it at all, so nothing->1->2 higher than 1 on local device